caching.py

```
@@ -0,0 +1,613 @@
+        self.cleanup_thread = None
+        self.running = True
+        self._start_cleanup_thread()
+    
+    def _start_cleanup_thread(self):
+        """Start background cleanup thread."""
+        def cleanup_worker():
+            while self.running:
+                try:
+                    time.sleep(60)  # Run every minute
+                    self._cleanup_expired()
+                except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)